AutomatedTesting/Gem/PythonTests/largeworlds/dyn_veg/EditorScripts/ShapeIntersectionFilter_FilterStageToggle.py

```diff
@@ -0,0 +1,113 @@
+"""
+Copyright (c) Contributors to the Open 3D Engine Project.
+For complete copyright and license terms please see the LICENSE at the root of this distribution.
+
+SPDX-License-Identifier: Apache-2.0 OR MIT
+"""
+
+
+class Tests:
+    instance_count_in_box_shape = (
+        "Only found instances in the configured Box Shape intersection area",
+        "Found instances outside of the configured Box Shape intersection area"
+    )
+    instance_count_in_cylinder_shape = (
+        "Only found instances in the configured Cylinder Shape intersection area",
+        "Found instances outside of the configured Cylinder Shape intersection area"
+    )
+    preprocess_instance_count = (
+        "Found the expected number of instances with preprocessing filter stage",
+        "Found an unexpected number of instances with preprocessing filter stage"
+    )
+    postprocess_instance_count = (
+        "Found the expected number of instances with postprocessing filter stage",
+        "Found an unexpected number of instances with postprocessing filter stage"
+    )
+
+
+def ShapeIntersectionFilter_FilterStageToggle():
+    """
+    Summary:
+    Filter Stage toggle affects final vegetation position
+
+    Expected Result:
+    Vegetation instances plant differently depending on the Filter Stage setting. With PreProcess, some vegetation
+    instances can appear on slopes outside the filtered values. With PostProcess, vegetation instances only appear on
+    the correct slope values.
+
+    :return: None
+    """
+
+    import os
+
+    import azlmbr.math as math
+    import azlmbr.legacy.general as general
+
+    import editor_python_test_tools.hydra_editor_utils as hydra
+    from largeworlds.large_worlds_utils import editor_dynveg_test_helper as dynveg
+    from editor_python_test_tools.utils import Report
+    from editor_python_test_tools.utils import TestHelper as helper
+
+    # Open an existing simple level
+    helper.init_idle()
+    helper.open_level("Physics", "Base")
+
+    general.set_current_view_position(512.0, 480.0, 38.0)
+
+    # Create basic vegetation entity
+    position = math.Vector3(512.0, 512.0, 32.0)
+    asset_path = os.path.join("Slices", "PinkFlower.dynamicslice")
+    vegetation = dynveg.create_vegetation_area("vegetation", position, 16.0, 16.0, 16.0, asset_path)
```

Comment:
Just a random note - we should probably start using prefabs for any new tests written.  I believe the code can successfully spawn both dynamic slices _and_ prefabs regardless of whether or not prefabs are enabled for the project.

Reply:
Agreed. This test isn't actually new, it was misnamed previously so went through a rename, but will definitely keep that in mind for new tests, and plan to update all of these to use prefabs in the near future.